File: server-dot-net/Model/Invoice.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace server_dot_net.Model
{
   // [Table("invoice")]
    public class Invoice
    {
        [Key]
        [Column("id")]
        public long Id { get; set; }

        [Required]

        [Column("invoice_number")]
        public string InvoiceNumber { get; set; }//  = string.Empty;

        [Column("amount")]
        public double Amount { get; set; }

        [Column("issue_date")]
        public DateTime IssueDate { get; set; }
        [Column("due_date")]
        public DateTime DueDate { get; set; }

        [ForeignKey("Merchant")]
        [Column("merchant_id")]
        public long MerchantId { get; set; }

        public Merchant Merchant { get; set; }

        [ForeignKey("Customer")]
        [Column("customer_id")]
        public long CustomerId { get; set; }

        public Customer Customer { get; set; }

        public string Status { get; set; } // Paid, Unpaid, Overdue
    }
}

File: server-spring-boot/src/main/java/com/finflow/server/entity/Invoice.java
package com.finflow.server.entity;

import jakarta.persistence.*;
import lombok.*;

import java.time.LocalDate;

@Entity
@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class Invoice {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    private String invoiceNumber;

    private Double amount;
    private LocalDate issueDate;
    private LocalDate dueDate; 

    @ManyToOne
    @JoinColumn(name = "merchant_id")
    private Merchant merchant;

    @ManyToOne
    @JoinColumn(name = "customer_id")
    private Customer customer;

    private String status; //  "Paid", "Unpaid", "Overdue"


    
}
